### ISSUE TITLE
--model foo panics, should show downloaded models and/or suggest a model instead.

### ISSUE THREAD

**Issue by gmatht** [TabbyML/tabby#3396]
If the user specifies an invalid model or no model at all Tabby could display models that are downloaded. If no models are downloaded it could suggest a simple default like StarCoder-1B

A similar feature could also be implemented for `--chat-model` too.

**Additional context**

At the moment Tabby just panics if a Invalid model_id is specified.

```
> .\tabby serve --model foo
The application panicked (crashed).
Message:  Invalid model_id <TabbyML/foo>
Location: crates\tabby-common\src\registry.rs:169

  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ BACKTRACE ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
   1: git_filter_source_repo<unknown>
      at <unknown source file>:<unknown line>
   2: git_filter_source_repo<unknown>
      at <unknown source file>:<unknown line>
   3: git_filter_source_repo<unknown>
      at <unknown source file>:<unknown line>
   4: git_filter_source_repo<unknown>
      at <unknown source file>:<unknown line>
   5: git_filter_source_mode<unknown>
      at <unknown source file>:<unknown line>
   6: git_filter_source_mode<unknown>
      at <unknown source file>:<unknown line>
   7: git_filter_source_mode<unknown>
      at <unknown source file>:<unknown line>
   8: git_filter_source_mode<unknown>
      at <unknown source file>:<unknown line>
   9: git_credential_userpass_plaintext_new<unknown>
      at <unknown source file>:<unknown line>
  10: git_odb_object_data<unknown>
      at <unknown source file>:<unknown line>
  11: git_filter_source_repo<unknown>
      at <unknown source file>:<unknown line>
  12: git_filter_source_repo<unknown>
      at <unknown source file>:<unknown line>
  13: git_filter_source_repo<unknown>
      at <unknown source file>:<unknown line>
  14: git_filter_source_repo<unknown>
      at <unknown source file>:<unknown line>
  15: git_filter_source_repo<unknown>
      at <unknown source file>:<unknown line>
  16: git_filter_source_repo<unknown>
      at <unknown source file>:<unknown line>
  17: git_filter_source_repo<unknown>
      at <unknown source file>:<unknown line>
  18: git_filter_source_mode<unknown>
      at <unknown source file>:<unknown line>
  19: git_filter_source_repo<unknown>
      at <unknown source file>:<unknown line>
  20: git_credential_userpass_plaintext_new<unknown>
      at <unknown source file>:<unknown line>
  21: BaseThreadInitThunk<unknown>
      at <unknown source file>:<unknown line>
  22: RtlUserThreadStart<unknown>
      at <unknown source file>:<unknown line>


```

---
Please reply with a 👍 if you want this feature.

**Comment by wsxiaoys** [comment]
related https://github.com/TabbyML/tabby/discussions/3285#discussioncomment-11196262

**Comment by 54corbin** [comment]
@wsxiaoys Can I take this issue?

**Comment by zwpaper** [comment]
Hi @54corbin, thank you for your willingness to tackle this issue. However, I apologize as we have resolved it in v0.22.0 and neglected to update the issue status. Perhaps you could find another issue to work on from the list of open 'good first issues' at https://github.com/TabbyML/tabby/issues?q=sort%3Aupdated-desc+is%3Aopen+label%3A%22good+first+issue%22.

![CleanShot X 2024-12-29 02 15 02](https://github.com/user-attachments/assets/852316e5-fb38-489f-be4a-96da77b10608)
